src/types/timezone.jl

```diff
@@ -1,27 +1,43 @@
-# Thread-local TimeZone cache, which caches time zones _per thread_, allowing thread-safe
+# Thread-local TimeZone caches, which caches time zones _per thread_, allowing thread-safe
 # caching. Note that this means the cache will grow in size, and may store redundant objects
 # accross multiple threads, but this extra space usage allows for fast, lock-free access
 # to the cache, while still being thread-safe.
-const THREAD_TZ_CACHES = Vector{Dict{String,Tuple{TimeZone,Class}}}()
+# Use a separate cache for FixedTimeZone (which is `isbits`) so the containers is concretely
+# typed and we avoid allocating a FixedTimeZone every time we get one from the cache.
+const THREAD_FTZ_CACHES = Vector{Dict{String,Tuple{FixedTimeZone,Class}}}()
+const THREAD_VTZ_CACHES = Vector{Dict{String,Tuple{VariableTimeZone,Class}}}()
```

Comment:
Since you're over here touching all of this code again, we _could_ consider switching to use https://github.com/JuliaConcurrent/MultiThreadedCaches.jl, which wraps up the pattern we used here into a standalone package. And that other package benefits from some eyes from `@tkf` and other engineers at RAI (@kpamnany and others), as well as some later attention to improving perf on accesses.

I don't want to complicate your PR too much, but I had been meaning to come back here and consider replacing the cache here with one from that package. I think that might be a nice cleanup since it would dramatically reduce the duplication here by using the factored out versions! :)